features/include/pcl/features/good.h

```diff
@@ -0,0 +1,297 @@
+	* \param[in] sign either 1 or -1.
+	* \param[out] histogram a 2D vector of unsigned int.
+	*/  
+      void 
+      create2DHistogramFromProjection (const PointCloudInPtr &projected_view, const float largest_side, 
+					const int8_t axis_a, const int8_t axis_b,
+					std::vector<std::vector<unsigned int> > &histogram) const;
+             
+      /** \brief normalizing a 1D histogram
+ 	* \param[in] histogram 1D histogram (int).
+	* \param[out] normalized_histogram the resultant normalized_histogram (float).
+        */       
+      void
+      normalizeHistogram (const std::vector <unsigned int> histogram, std::vector <float> &normalized_histogram);
+
+      /** \brief compute viewpoint entropy used for concatenating projections
+ 	* \param[in] normalized_histogram normalized_histogram (float).
+	* \param[out] entropy the resultant view entropy.
+        */   
+      void
+      viewpointEntropy (const std::vector <float> normalized_histogram, float &entropy);
+
+      /** \brief find max view point entropy used for concatenating projections
+ 	* \param[in] view_point_entropy a vector of float contains three view entropies.
+	* \param[out] index the resultant projection index.
+        */  
+      void
+      findMaxViewPointEntropy (const std::vector<float> view_point_entropy, int &index);
+
+      /** \brief compute variance of a given histogram
+	* \param[in] histogram normalized histogram of projection.
+	* \param[out] variance the resultant variance.
+        */        
+      void
+      varianceOfHistogram (const std::vector<float> histogram, float &variance);
+
+      /** \brief concatinating three orthographic projections
+       * three histograms are obtained for the projections; afterwards, two statistic features including entropy and variance have been calculated for each distribution vector;
+       * the histograms are consequently concatenated together using entropy and variance features, to form a single description for the given object. The ordering of the three
+       * histograms is first by decreasing values of entropy. Afterwards the second and third vectors are sorted again by increasing values of variance.
+	* \param[in] maximum_entropy_index index of orthographic projection that has maximum entropy.		
+	* \param[in] normalized_projected_views a vector of vector of float contains three normalized histogram of projected views
+	* \param[out] sorted_normalized_projected_views a vector of float representing the GOOD description for a set of points given by setInputCloud() 
+	* \param[out] name_of_sorted_projected_plane an string represents the order of concatenating projections.
+        */              
+      void
+      objectViewHistogram (int maximum_entropy_index, const std::vector<std::vector<float> > normalized_projected_views,
+		      std::vector<float> &sorted_normalized_projected_views,
+		      std::string &name_of_sorted_projected_plane /*debug*/);

```

Comment:
- One parameter per line
- normalized_projected_views needs to be passed by reference